src/main/java/seedu/address/logic/commands/DeleteInfoCommand.java

```diff
@@ -0,0 +1,134 @@
+
+
+    //TODO test cases
+    /**
+     * Creates a DeleteInfoCommand to delete the specified {@code Person}'s information
+     */
+    public DeleteInfoCommand(Nric targetNric, boolean[] fieldsToDelete) {
+        requireNonNull(targetNric);
+        requireNonNull(fieldsToDelete);
+        this.targetNric = targetNric;
+        this.fieldsToDelete = fieldsToDelete;
+    }
+
+    //TODO test cases
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+
+        ObservableList<Person> persons = model.getFilteredPersonList();
+        if (!model.hasPerson(Person.createPersonWithNric(targetNric))) {
+            throw new CommandException(Messages.MESSAGE_PERSON_NOT_FOUND);
+        }
+        //Difference between filteredPersons.contains and model.hasPerson: first checks if the instance is in the list,
+        //second checks if the NRIC is in the list
+        Person personToChange = persons.filtered(person -> person.getNric().equals(targetNric)).get(0);
+        if (fieldsToDelete[Fields.EMAIL.ordinal()]) {
+            personToChange.setEmail(null);
+        }
+        if (fieldsToDelete[Fields.ALLERGIES.ordinal()]) {
+            personToChange.setAllergies(null);
+        }
+        if (fieldsToDelete[Fields.BLOODTYPE.ordinal()]) {
+            personToChange.setBloodType(null);
+        }
+        if (fieldsToDelete[Fields.DATEOFADMISSION.ordinal()]) {
+            personToChange.setDateOfAdmission(null);
+        }
+        if (fieldsToDelete[Fields.COUNTRY.ordinal()]) {
+            personToChange.setCountry(null);
+        }
+        if (fieldsToDelete[Fields.CONDITION.ordinal()]) {
+            personToChange.setCondition(null);
+        }
+        if (fieldsToDelete[Fields.SYMPTOM.ordinal()]) {
+            personToChange.setSymptom(null);
+        }
+        if (fieldsToDelete[Fields.DIAGNOSIS.ordinal()]) {
+            personToChange.setDiagnosis(null);
+        }
+        return new CommandResult(String.format(MESSAGE_DELETE_PERSON_INFORMATION_SUCCESS,
+                Messages.format(personToChange)));
+    }
+
+    private boolean fieldsToDeleteEquals(boolean[] otherFieldsToDelete) {
+        if (otherFieldsToDelete.length != NUM_FIELDS) {
+            return false;
+        }
+        for (int i = 0; i < NUM_FIELDS; i++) {
+            if (this.fieldsToDelete[i] != otherFieldsToDelete[i]) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        if (other == this) {
+            return true;
+        }
+
+        // instanceof handles nulls
+        if (!(other instanceof DeleteInfoCommand)) {
+            return false;
+        }
+
+        DeleteInfoCommand otherDeleteCommand = (DeleteInfoCommand) other;
+        return targetNric.equals(otherDeleteCommand.targetNric)
+                && fieldsToDeleteEquals(otherDeleteCommand.fieldsToDelete);
+    }
+
```

Comment:
Remember to override hashCode() as well whenever you override equals()

Reply:
added optional fields in hashcode